tests/components/google_travel_time/test_config_flow.py

```diff
@@ -136,6 +136,181 @@ async def test_malformed_api_key(hass: HomeAssistant) -> None:
     assert result2["errors"] == {"base": "invalid_auth"}
 
 
+@pytest.mark.parametrize(
+    ("data", "options"),
+    [
+        (
+            MOCK_CONFIG,
+            {
+                CONF_MODE: "driving",
+                CONF_UNITS: UNITS_IMPERIAL,
+            },
+        )
+    ],
+)
+@pytest.mark.usefixtures("validate_config_entry", "bypass_setup")
+async def test_reconfigure(hass: HomeAssistant, mock_config) -> None:
+    """Test reconfigure flow."""
+    reconfigure_result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={
+            "source": config_entries.SOURCE_RECONFIGURE,
+            "entry_id": mock_config.entry_id,
+        },
+    )
+    assert reconfigure_result["type"] is FlowResultType.FORM
+    assert reconfigure_result["step_id"] == "reconfigure"
+
+    reconfigure_successful_result = await hass.config_entries.flow.async_configure(
+        reconfigure_result["flow_id"],
+        RECONFIGURE_CONFIG,
+    )
+    assert reconfigure_successful_result["type"] is FlowResultType.ABORT
+    assert reconfigure_successful_result["reason"] == "reconfigure_successful"
+    await hass.async_block_till_done()
+
+    entry = hass.config_entries.async_entries(DOMAIN)[0]
+    assert entry.data == {
+        CONF_API_KEY: "api_key2",
+        CONF_ORIGIN: "location3",
+        CONF_DESTINATION: "location4",
+    }
+
+
+@pytest.mark.parametrize(
+    ("data", "options"),
+    [
+        (
+            MOCK_CONFIG,
+            {
+                CONF_MODE: "driving",
+                CONF_UNITS: UNITS_IMPERIAL,
+            },
+        )
+    ],
+)
+@pytest.mark.usefixtures("invalidate_config_entry")
+async def test_reconfigure_invalid_config_entry(
```

Comment:
These failure tests should end with `abort` as in a successful reconfigure.

Reply:
Why should they be a successful reconfiguration when the user supplied wrong information?

Reply:
It should not. It should continue by the user setting the right information within the same flow eg the user try again.

Reply:
The current behavior is that if the user enters wrong information (e.g. api wrong) or the api is unresponsive he will be presented with the same flow again and the error is displayed.

Do you rather want that the flow gets aborted and the user has to click `Reconfigure` on the ConfigEntry again?

Reply:
That is how it currently works

Reply:
This test and below tests are all ending with a form which is presented to the user with an error. 
I want all these tests to continue from that point with the user entering data again and then it's successful eg it ends with an `abort`

All config entry tests should end with either `abort` or `create_entry` (so the tests above these are actually also not correct). 

Will provide some references tomorrow if still unclear what I mean.

Reply:
We should test that for any type of exception the flow can recover from those exceptions and continue until either an entry is created or we abort the flow.

Reply:
I see you ignored my comment. I will add additional test cases to get this merged....

Reply:
I'm not sure I understand what you mean.
In the test cases with an exception you end them with a `form` which means the flows are still open.
I don't want to have more tests, I want the current tests extended so they finish properly.

We want such test cases to continue from that point so we remove the exception, allow the user to again enter the form data so it can end in a success (`create_entry` or `abort`).
By this the flows ends properly and we also see that a config flow can recover from exceptions so a user would not have to restart a flow but can simply again put in form data and press submit.

[Here](https://github.com/home-assistant/core/blob/dev/tests/components/sensibo/test_config_flow.py#L63-L122) is an example what I mean by a recover from a failure so in this case it ends with `create_entry`